### PR TITLE
Added AMQP connection error handler.

### PR DESCRIPTION
**Pull request by jlongever**

### CODE REVIEW

**Comment by jlongever** [comment]
@RackHD/corecommitters @zyoung51 @uppalk1 @tannoa2

File: test/modules/amqp.py [hunk 1]
```diff
@@ -40,6 +44,14 @@ def on_message(self, body, message):
     def on_conn_retry(self):
         LOG.error('Retrying connection for {0}'.format(self.__amqp_url))
 
+    def on_connection_error(self, exc, interval):
```
Comment (zyoung51) [inline test/modules/amqp.py]: If this is called to re-establish a connection, should the __max_error be reset once the connection is established and good again?

Comment (jlongever) [inline test/modules/amqp.py]: Good point.  Maybe I can call ensure_connection() and reset __max_error if the connection good?

Comment (jlongever) [inline test/modules/amqp.py]: nvm, ensure_connection takes the on_error callback.

Comment (jlongever) [inline test/modules/amqp.py]: @zyoung51 The connection retry should be reset by ensure_connection when the connection is reestablished (ensure_connection maintains it's own retry count and interval.)   The __max_error is a global threshold for the number of ensure_connection attempts.

**Comment by benbp** [comment]
:+1:

**Comment by keedya** [comment]
:+1:

**Pull request merged**